pages/signin.tsx

```diff
@@ -0,0 +1,156 @@
+import styles from "../styles/signin.module.css";
+import { GitHubIcon } from "../components/GitHubIcon";
+import * as React from 'react';
+import LoadingButton from '@mui/lab/LoadingButton';
+import TrafficIcon from '@mui/icons-material/Traffic';
+import config from "../config.json"
+
+
+const SCOPE = "repo"
+
+// get code
+function redirect() {
+	const uuid = self.crypto.randomUUID();
+	window.location.href = `https://github.com/login/oauth/authorize?client_id=${config.clientID}&redirect_uri=${config.redirectURL}&scope=${SCOPE}&state=${uuid}`;
+}
+
+type SignInStatus = "calculating" | "not-signed-in" | "exchanging-code-for-token" | "signed-in" | "error-while-signing-in"
+
+
+export default function SignIn() {
+
+	const router = useRouter()
+
+	const [token, setToken] = useState<string>('');
+
+	const [signInStatus, setSignInStatus] = useState<SignInStatus>("calculating")
+
+	const [errorMessage, setErrorMessage] = useState<string>()
+
+	useEffect(() => {
+		setErrorMessage(undefined as any)
+
+		if (router == null) {
+			return
+		}
+		const code = router.query.code?.toString()
+		const error = router.query.error?.toString()
+
+		if (error != null) {
+			setSignInStatus('error-while-signing-in')
+			error === "access_denied" && setErrorMessage("You have denied access to your GitHub account. Please try again.")
+		} else if (code != null) {
+			setSignInStatus('exchanging-code-for-token')
+			console.log(`code ${code}`);
+
+			//This immediately called function must be defined because useEffect handlers cannot be async
+			const isSignedIn = async () => {
+				const response = await fetch(`/api/authenticate/${encodeURIComponent(code.toString())}/`);
+				if (response.status == 200) {
+					return true
+				}
+
+				throw await response.json()
+			};
+			isSignedIn()
+				.then(async () => {
+					setSignInStatus('signed-in')
+				})
+				.catch((error) => {
+
+					console.log("error while exchanging token", error)
+					setSignInStatus('error-while-signing-in')
+					error?.message && setErrorMessage(error.message)
+
+				});
+
+		} else {
+			setSignInStatus('not-signed-in')
+		}
+
+	}, [router]);
+
+	/**
+	 * Called when the user clicks the Try Again button after an error occurred
+	 */
+	function handleTryAgain() {
+		router.push("/signin")
+	}
+
+
+	return (
+		<>
+			{/* <button onClick={redirect}>Login</button> */}
+			<Grid container className={styles.signinPage} justifyContent='center' alignContent='center'>
+				<Paper className={styles.signinContainer}>
+					{/* TODO: evergreen dashborad logo, we are aming for this design https://gitter.im/ */}
+					<Typography variant="h5" component="h1"><span className={styles.everGreen}>Evergreen Dashboard</span> </Typography>
+					<Typography variant="body1">You can read <Link href="https://github.com/ahm-monash/evergreen/documentation/oauth-scope.md">our documentation</Link> on OAuth scopes to see why we request certain OAuth scopes.</Typography>
```

Comment:
not sure if this is a config issue on my end, but this link doesn't work for me - I get this 'Not Found' 
![image](https://user-images.githubusercontent.com/81656054/189237644-5d983dd2-5f39-4085-bd18-260047973518.png)

Reply:
I just saw the TODO whoops, maybe remove link until it is working?

Reply:
Yep we haven't setup any documentation page for this yet. We mainly concentrated on the functionality rather than the UI. I'll remove the link